cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake

```diff
@@ -6,126 +6,176 @@ macro(SetupShibokenAndPyside)
+                                    "#QCoreApplication.CodecForTr=0\n"
+                                    "#QCoreApplication.UnicodeUTF8=1\n")
+        file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtGui.py  "from PySide2.QtGui import *\n"
+                                    "from PySide2.QtWidgets import *\n"
+                                    "QHeaderView.setResizeMode = QHeaderView.setSectionResizeMode\n")
+        file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtSvg.py  "from PySide2.QtSvg import *\n")
+        file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtUiTools.py  "from PySide2.QtUiTools import *\n")
+        file(WRITE ${CMAKE_BINARY_DIR}/Ext/PySide/QtWidgets.py  "from PySide2.QtWidgets import *\n")
+
+        if(APPLE AND NOT BUILD_WITH_CONDA)
+            install(
+            DIRECTORY
+                ${CMAKE_BINARY_DIR}/Ext/PySide
+            DESTINATION
+                MacOS
+            )
+        else()
+            install(
+            DIRECTORY
+                ${CMAKE_BINARY_DIR}/Ext/PySide
+            DESTINATION
+                Ext
+            )
+        endif()
 
-    # If shiboken cannot be found the build option will be set to OFF
-    if(SHIBOKEN_INCLUDE_DIR)
-        option(FREECAD_USE_SHIBOKEN "Links to the shiboken library at build time. If OFF its Python module is imported at runtime" ON)
-    else()
-        option(FREECAD_USE_SHIBOKEN "Links to the shiboken library at build time. If OFF its Python module is imported at runtime" OFF)
+        # If shiboken cannot be found the build option will be set to OFF
+        if(SHIBOKEN_INCLUDE_DIR)
+            option(FREECAD_USE_SHIBOKEN "Links to the shiboken library at build time. If OFF its Python module is imported at runtime" ON)
+        else()
+            option(FREECAD_USE_SHIBOKEN "Links to the shiboken library at build time. If OFF its Python module is imported at runtime" OFF)
+
+            # Now try to import the shiboken Python module and print a warning if it can't be loaded
+            execute_process(
+            COMMAND ${PYTHON_EXECUTABLE} -c "import shiboken2"
+            RESULT_VARIABLE FAILURE
+            OUTPUT_VARIABLE PRINT_OUTPUT
+            )
+
+            if(FAILURE)
+                message("=================================\n"
+                        "shiboken Python module not found.\n"
+                        "=================================\n")
+            endif()
+        endif()
+
+        # If PySide cannot be found the build option will be set to OFF
+        if(PYSIDE_INCLUDE_DIR)
+            option(FREECAD_USE_PYSIDE "Links to the PySide libraries at build time." ON)
+        else()
+            option(FREECAD_USE_PYSIDE "Links to the PySide libraries at build time." OFF)
+        endif()
 
-        # Now try to import the shiboken Python module and print a warning if it can't be loaded
+        # Independent of the build option PySide modules must be loaded at runtime. Print a warning if it fails.
```

Comment:
It turns out it's not true (Werner pointed out in the forums that we really are supposed to be able to compile and run without PySide/Shiboken) -- so I'll have to go back in and fix this.

Reply:
(Misunderstood, let me rewrite this comment...) I can't use the "required" there because if pip was used to install PySide, `find_package` won't find it. So the only check I can actually do is to try to run some code that imports PySide and see if it works.